packages/typespec-autorest/test/models.test.ts

```diff
@@ -665,6 +665,45 @@ describe("typespec-autorest: model definitions", () => {
         },
       });
     });
+
+    it("defines nullable model", async () => {
+      const res = await openApiFor(
+        `
+        model Pet {
+          name: string;
+        }
+  
+        model Dog {
+          type: Pet | null;
+        }
+        `,
+      );
+      deepStrictEqual(res.definitions.Dog.properties.type, {
+        $ref: "#/definitions/Pet",
+        "x-nullable": true,
+      });
+    });
+
+    it("defines nullable record", async () => {
```

Comment:
Should we have a test for array types as well (if we don't already)?

`arrayProp: Array<Pet | null>;`